core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -7,11 +7,20 @@ class SemiCollapsibleNavigationClass {
     this.responsiveNavSetting = LuigiConfig.getConfigValue(
       'settings.responsiveNavigation'
     );
-    this.semiCollapsible =
+    let currentCollapsedState = localStorage.getItem(
+      NavigationHelpers.COL_NAV_KEY
+    );
+    let getResponsiveNavSettings =
       this.responsiveNavSetting === 'semiCollapsible' ||
-      this.responsiveNavSetting === 'Fiori3'
-        ? true
-        : false;
+      this.responsiveNavSetting === 'Fiori3';
+
+    //checking if here was a previous state in localStorage before the reload
+    if (currentCollapsedState != false && getResponsiveNavSettings) {
```

Comment:
I would remove the else part, because after a reload the variable `this.semiCollapsible` will be always undefined and this lead to problems if you resize from desktop to mobile.
Code could look like this:

```
//checking if here was a previous state in localStorage before the reload
    if (currentCollapsedState !== null && getResponsiveNavSettings) {
      this.isSemiCollapsed = this.getCollapsed();
    }
    this.semiCollapsible = getResponsiveNavSettings ? true : false;
```

Reply:
done, thanks for checking